sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncyptedBlockBlobAPITest.groovy

```diff
@@ -1462,6 +1468,32 @@ class EncyptedBlockBlobAPITest extends APISpec {
         3 * Constants.MB| Constants.MB  || 4 // Encryption padding will add an extra block
     }
 
+    def getPerCallVersionPolicy() {
+        return new HttpPipelinePolicy() {
+            @Override
+            Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+                context.getHttpRequest().setHeader("x-ms-version","2017-11-09")
+                return next.process()
+            }
+            @Override
+            HttpPipelinePosition getPipelinePosition() {
+                return HttpPipelinePosition.PER_CALL
+            }
+        }
+    }
+
+    def "Per call policy"() {
+        def client = getEncryptedClientBuilder(fakeKey, fakeKeyResolver, primaryCredential, bec.getBlobUrl(), getPerCallVersionPolicy()).buildEncryptedBlobClient()
```

Comment:
Can you leave a comment that explains this tests the policy is in the right place because if it were added per retry, it would be after the credentials and auth would fail because we changed a signed header

Reply:
done